### PR TITLE
PDS-612 Adding custom rows per page

### PR DESCRIPTION
**Pull request by cathal41**
### Description of proposed changes
Add a new prop to the TableFooter to allow the supply of a custom list of rows per page

### Screenshot of proposed changes
N/A

### CODE REVIEW

File: packages/data-grid/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@puppet/data-grid",
-  "version": "0.6.10",
+  "version": "0.6.11",
```
Comment (mardotio) [inline packages/data-grid/package.json:3]: I think you'll also need to update this value in the lock file

File: packages/data-grid/src/tableFooter/TableFooter.jsx [hunk 2]
```diff
@@ -129,7 +131,18 @@ function TableFooter({
...
 }) {
+  const rowsPerPageOptionValues =
+    rowsPerPageOptions.length > 0
+      ? rowsPerPageOptions
+      : [
+          { value: 5, label: '5' },
+          { value: 10, label: '10' },
+          { value: 25, label: '25' },
+          { value: 50, label: '50' },
+        ];
```
Comment (mardotio) [inline packages/data-grid/src/tableFooter/TableFooter.jsx:144]: Could you move this to be defined in `defaultProps`? I'm pretty sure that's how the default values shown in design system website get populated.

**Review by mardotio — COMMENTED** [review]
Looks good, just a couple of minor things.

Could you also squash your commits so that we don't merge too much noise into main?

File: packages/data-grid/src/tableFooter/TableFooter.jsx (hunk 2 above)
Comment (cathal41) [reply packages/data-grid/src/tableFooter/TableFooter.jsx:144]: The defaultProps are used to populate the props table at the top of the design library page - https://puppet.style/#/Data%20Grid/TableFooter
Adding this into the default props will not be formatted correctly for the page. I tried this and it try's to force it into the space.

**Comment by cathal41** [comment]
> Looks good, just a couple of minor things.
> 
> Could you also squash your commits so that we don't merge too much noise into main?

I'll do a squash merge when the PR is done

**Review by mardotio — APPROVED** [review]

**Pull request merged**